<h2><a href="https://github.com/imjiaoyuan/imjiaoyuan.github.io/blob/master/content/p/tbtools-completes-paper-reproduction.md">仓库源文</a>，<a href="https://yuanj.top/tbtools-completes-paper-reproduction">站点原文</a></h2>
<p>文章题为"Identification and expression pattern analysis of the <em>OsSnRK2</em> gene family in rice"。该文章发表于 Frontiers in Plant Science（中科院二区，IF=5.6，第一作者：Tongyuan Yu，通讯作者：Dawei Xue）。</p>
&lt;!--more--&gt;

<p>根据原文的材料与方法，文章做了以下生信分析：</p>
<ul>
<li>domain 分析</li>
<li>染色体定位</li>
<li>基因结构分析</li>
<li>motif 分析</li>
<li>系统发育树</li>
<li>共线性分析</li>
<li>多序列比对</li>
<li>蛋白质理化性质分析</li>
<li>蛋白质二、三级结构预测</li>
<li>互作蛋白网络</li>
<li>顺式作用元件分析</li>
<li>表达热图</li>
</ul>
<p>以上的分析基本都可以用 TBtools 来完成并做成好看的图片，这也是一个常规基因家族分析的流程。此处不涉及过于详细的步骤，更详细的步骤请参阅我的博客文章：<a href="https://yuanj.top/p/gene-family-identification/">https://yuanj.top/p/gene-family-identification/</a></p>
<p>作者在补充材料里提供了基因 ID、基因名和蛋白序列：</p>
<pre><code class="lang-txt">&gt;OsSAPK1
MERYEVMRDIGSGNFGVAKLVRDVATNHLFAVKFIERGLKIDEHVQREIMNHRSLKHPNIIRFKEVVLTPTHLAIVMEYAAGGELFERICNAGRFSEDEARFFFQQLISGVSYCHSMQVCHRDLKLENTLLDGSVTPRLKICDFGYSKSSVLHSQPKSTVGTPAYIAPEVLSRKEYDGKVADVWSCGVTLYVMLVGAYPFEDPDDPRNFRKTITRILSVQYSIPDYVRVSADCRHLLSRIFVGNPEQRITIPEIKNHPWFLKNLPIEMTDEYQRSMQLADMNTPSQSLEEVMAIIQEARKPGDAMKLAGAGQVACLGSMDLDDIDDIDDIDIENSGDFVCAL
&gt;OsSAPK2
MERYEVIKDIGSGNFGVAKLVRDVRTKELFAVKFIERGQKIDENVQREIMNHRSLRHPNIVRFKEVVLTPTHLAIVMEYAAGGELFERICSAGRFSEDEARFFFQQLISGVSYCHSMQICHRDLKLENTLLDGSIAPRLKICDFGYSKSSLLHSQPKSTVGTPAYIAPEVLARKEYDGKVADVWSCGVTLYVMLVGAYPFEDPDEPRNFRKTITRILSVQYMVPDYVRVSMECRHLLSRIFVANPEQRITIPEIKNHPWFLKNLPIEMTDEYQMSVQMNDINTPSQGLEEIMAIIQEARKPGDGSKFSGQIPGLGSMELDDVDTDDIDVEDSGDFVCAL
&gt;OsSAPK3
MEERYEALKELGAGNFGVARLVRDKRSKELVAVKYIERGKKIDENVQREIINHRSLRHPNIIRFKEVCLTPTHLAIVMEYAAGGELFEQICTAGRFSEDEARYFFQQLISGVSYCHSLEICHRDLKLENTLLDGSPTPRVKICDFGYSKSALLHSKPKSTVGTPAYIAPEVLSRKEYDGKATDVWSCGVTLYVMLVGSYPFEDPGDPRNFRKTISRILGVQYSIPDYVRVSSDCRRLLSQIFVADPSKRITIPEIKKHTWFLKNLPKEISEREKADYKDTDAAPPTQAVEEIMRIIQEAKVPGDMAAADPALLAELAELKSDDEEEAADEYDTY
&gt;OsSAPK4
MEKYEAVRDIGSGNFGVARLMRNRETRELVAVKCIERGHRIDENVYREIINHRSLRHPNIIRFKEVILTPTHLMIVMEFAAGGELFDRICDRGRFSEDEARYFFQQLICGVSYCHHMQICHRDLKLENVLLDGSPAPRLKICDFGYSKSSVLHSRPKSAVGTPAYIAPEVLSRREYDGKLADVWSCGVTLYVMLVGAYPFEDQDDPKNIRKTIQRIMSVQYKIPDYVHISAECKQLIARIFVNNPLRRITMKEIKSHPWFLKNLPRELTETAQAMYYRRDNSVPSFSDQTSEEIMKIVQEARTMPKSSRTGYWSDAGSDEEEKEEEERPEENEEEEEDEYDKRVKEVHASGELRMSSLRI
&gt;OsSAPK5
MEKYEPVREIGAGNFGVAKLMRNKETRELVAMKFIERGNRIDENVFREIVNHRSLRHPNIIRFKEVVVTGRHLAIVMEYAAGGELFERICEAGRFHEDEARYFFQQLVCGVSYCHAMQICHRDLKLENTLLDGSPAPRLKICDFGYSKSSLLHSRPKSTVGTPAYIAPEVLSRREYDGKLADVWSCGVTLYVMLVGAYPFEDPKDPKNFRKTISRIMSVQYKIPEYVHVSQPCRHLLSRIFVANPYKRISMGEIKSHPWFLKNLPRELKEEAQAVYYNRRGADHAASSASSAAAAAAFSPQSVEDIMRIVQEAQTVPKPDKPVSGYGWGTDDDDDDQQPAEEEDEEDDYDRTVREVHASVDLDMSNLQIS
&gt;OsSAPK6
MEKYELLKDIGSGNFGVARLMRNRETKELVAMKYIPRGLKIDENVAREIINHRSLRHPNIIRFKEVVLTPTHLAIVMEYAAGGELFDRICSAGRFSEDESRYFFQQLICGVSYCHFMQICHRDLKLENTLLDGSPAPRLKICDFGYSKSSLLHSKPKSTVGTPAYIAPEVLSRREYDGKMADVWSCGVTLYVMLVGAYPFEDPDDPKNFRKTIGRIVSIQYKIPEYVHISQDCRQLLSRIFVANPAKRITIREIRNHPWFMKNLPRELTEAAQAKYYKKDNSARTFSDQTVDEIMKIVQEAKTPPPSSTPVAGFGWTEEEEQEDGKNPDDDEGDRDEEEGEEGDSEDEYTKQVKQAHASCDLQKS
&gt;OsSAPK7
MERYELLKDIGAGNFGVARLMRNKETKELVAMKYIPRGLKIDENVAREIINHRSLRHPNIIRFKEVVVTPTHLAIVMEYAAGGELFDRICNAGRFSEDEARYFFQQLICGVSYCHFMQICHRDLKLENTLLDGSPAPRLKICDFGYSKSSLLHSKPKSTVGTPAYIAPEVLSRREYDGKTADVWSCGVTLYVMLVGAYPFEDPDDPKNFRKTIGRIMSIQYKIPEYVHVSQDCRQLLSRIFVANPAKRITIREIRNHPWFLKNLPRELTEAAQAMYYKKDNSAPTYSVQSVEEIMKIVEEARTPPRSSTPVAGFGWQEEDEQEDNSKKPEEEQEEEEDAEDEYDKQVKQVHASGEFQLS
&gt;OsSAPK8
MAAAGAGAGAPDRAALTVGPGMDMPIMHDSDRYELVRDIGSGNFGVARLMRDRRTMELVAVKYIERGEKIDDNVQREIINHRSLKHPNIIRFKEVILTPTHLAIVMEYASGGELFERICKNVRFSEDEARYFFQQLISGVSYCHSMQVCHRDLKLENTLLDGSPAPRLKICDFGYSKSSVLHSQPKSTVGTPAYIAPEVLLKKEYDGKTADVWSCGVTLYVMVVGAYPFEDPEEPKNFRKTIQRILNVQYSIPENVDISPECRHLISRIFVGDPSLRITIPEIRSHGWFLKNLPADLMDDDSMSSQYEEPDQPMQTMDQIMQILTEATIPPACSRINHILTDGLDLDDDMDDLDSDSDIDVDSSGEIVYAM
&gt;OsSAPK9
MERAAAGPLGMEMPIMHDGDRYELVKEIGSGNFGVARLMRNRASGDLVAVKYIDRGEKIDENVQREIINHRSLRHPNIIRFKEVILTPTHLAIVMEYASGGELFERICSAGRFSEDEARFFFQQLISGVSYCHSMQVCHRDLKLENTLLDGSTAPRLKICDFGYSKSSVLHSQPKSTVGTPAYIAPEVLLKKEYDGKIADVWSCGVTLYVMLVGAYPFEDPEDPKNFRKTIQKILGVQYSIPDYVHISPECRDLITRIFVGNPASRITMPEIKNHPWFMKNIPADLMDDGMVSNQYEEPDQPMQNMNEIMQILAEATIPAAGTSGINQFLTDSLDLDDDMEDMDSDLDLDIESSGEIVYAM
&gt;OsSAPK10
MDRAALTVGPGMDMPIMHDGDRYELVRDIGSGNFGVARLMRSRADGQLVAVKYIERGDKIDENVQREIINHRSLRHPNIIRFKEVILTPTHLAIVMEYASGGELFERICNAGRFSEDEARFFFQQLISGVSYCHSMQVCHRDLKLENTLLDGSTAPRLKICDFGYSKSSVLHSQPKSTVGTPAYIAPEVLLKKEYDGKIADVWSCGVTLYVMLVGAYPFEDPDEPKNFRKTIQRILGVQYSIPDYVHISPECRDLIARIFVANPATRISIPEIRNHPWFLKNLPADLMDDSKMSSQYEEPEQPMQSMDEIMQILAEATIPAAGSGGINQFLNDGLDLDDDMEDLDSDPDLDVESSGEIVYAM
</code></pre>
<h2>domain 分析</h2>
<p>结构域（domain）是位于超二级结构和三级结构间的一个层次。结构域是在蛋白质的三级结构内的独立折叠单元，通常都是几个超二级结构单元的组合。在较大的蛋白质分子中，由于多肽链上相邻的超二级结构紧密联系，进一步折叠形成一个或多个相对独立的致密三维实体，即结构域。</p>
<p>结构域实质上是二级结构的组合体，充当三级结构的构件。每个结构域分别代表一种功能。</p>
<p>原文中基因家族是指水稻中一系列具有 S_TKc 结构域的基因，是磷酸转移酶，丝氨酸或苏氨酸特异性激酶亚家族。我们可以在 SMART 网站查看该结构域的信息：<a href="http://smart.embl.de/smart/do_annotation.pl?DOMAIN=SM00220">S_TKc</a>。补充材料里面也提供了基因的蛋白序列，我们直接拿过来用。将蛋白序列粘贴到 <a href="http://smart.embl.de/">SMART 网站</a> 查看结构域：</p>
<p><img alt="" src="/i/20240427092258.jpg"/></p>
<p>结果显示所有的基因都是具有 S_TKc 结构域的。但是个人感觉 SMART 网站的结果可能看上去不是很友好，我喜欢用 <a href="https://www.ncbi.nlm.nih.gov/Structure/bwrpsb/bwrpsb.cgi">NCBI CD-search</a>，也是上传序列开始分析即可，显示效果更加直观：</p>
<p><img alt="" src="/i/20240427092304.jpg"/></p>
<p>并且也可以用 TBtools 做成图：</p>
<p><img alt="" src="/i/20240427092386.jpg"/></p>
<h2>染色体定位</h2>
<p>从 <a href="https://plants.ensembl.org/Oryza_sativa/Info/Index">Ensembl plants</a> 下载水稻（日本晴）基因组的注释文件，然后将基因 ID 写在一个文件中：</p>
<pre><code class="lang-txt">transcript:Os03t0390200-01
transcript:Os07t0622000-01
transcript:Os10t0564500-01
transcript:Os01t0869900-01
transcript:Os04t0691100-01
transcript:Os02t0551100-01
transcript:Os04t0432000-01
transcript:Os03t0764800-01
transcript:Os12t0586100-01
transcript:Os03t0610900-01
</code></pre>
<p>再准备一个 rename 文件：</p>
<pre><code class="lang-txt">transcript:Os03t0390200-01  OsSAPK1
transcript:Os07t0622000-01  OsSAPK2
transcript:Os10t0564500-01  OsSAPK3
transcript:Os01t0869900-01  OsSAPK4
transcript:Os04t0691100-01  OsSAPK5
transcript:Os02t0551100-01  OsSAPK6
transcript:Os04t0432000-01  OsSAPK7
transcript:Os03t0764800-01  OsSAPK8
transcript:Os12t0586100-01  OsSAPK9
transcript:Os03t0610900-01  OsSAPK10
</code></pre>
<p>这里因为 Ensembl plants 下载的注释文件中默认的基因 ID 是转录本的，所以需要对 ID 进行一些处理，加上<code>transcript:</code>再把 Os 的 ID 转换成转录本 ID 即可。</p>
<p>导入到 TBtools：</p>
<p><img alt="" src="/i/20240427092353.jpg"/></p>
<p>出图后美化一下：</p>
<p><img alt="" src="/i/20240427092338.jpg"/></p>
<h2>基因结构分析&amp;motif 分析</h2>
<p>将蛋白序列上传到 <a href="https://meme-suite.org/meme/tools/meme">meme 网站</a> 后下载输出的 xml 文件即可，但是网页版 MEME Suite 经常要排队，所以可以用 TBtools 进行本地分析：</p>
<p><img alt="" src="/i/20240427092331.jpg"/></p>
<p>再将输出的 xml 文件可视化：</p>
<p><img alt="" src="/i/20240427092365.jpg"/></p>
<p>通常我们将基因结构、domain、motif 展示在一张图上：</p>
<p><img alt="" src="/i/20240427092443.jpg"/></p>
<p><img alt="" src="/i/20240427092488.jpg"/></p>
<h2>系统发育树</h2>
<p>作者将拟南芥和大麦中的 SnRK2 家族基因也放在一起进行了系统发育分析，我在文章"# Genome-wide identification and expression pattern of SnRK gene family under several hormone treatments and its role in floral scent emission in <em>Hedychium coronarium</em>"中找到了 10 个拟南芥 SnRK2 基因，在文章"Genome-Wide Identification, Expression Pattern and Sequence Variation Analysis of <em>SnRK</em> Family Genes in Barley"中找到 11 个大麦 SnRK2 基因，提取其蛋白质序列共同进行系统发育分析：</p>
<table>
<thead><tr>
<th style="text-align:left">Gene Name</th>
<th style="text-align:left">Gene ID</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left">AtSnRK1.1</td>
<td style="text-align:left">At3g01090</td>
</tr>
<tr>
<td style="text-align:left">AtSnRK1.2</td>
<td style="text-align:left">At3g29160</td>
</tr>
<tr>
<td style="text-align:left">AtSnRK1.3</td>
<td style="text-align:left">At5g39440</td>
</tr>
<tr>
<td style="text-align:left">AtSnRK2.1</td>
<td style="text-align:left">At5g08590</td>
</tr>
<tr>
<td style="text-align:left">AtSnRK2.2</td>
<td style="text-align:left">At3g50500</td>
</tr>
<tr>
<td style="text-align:left">AtSnRK2.3</td>
<td style="text-align:left">At5g66880</td>
</tr>
<tr>
<td style="text-align:left">AtSnRK2.4</td>
<td style="text-align:left">At1g10940</td>
</tr>
<tr>
<td style="text-align:left">AtSnRK2.5</td>
<td style="text-align:left">At5g63650</td>
</tr>
<tr>
<td style="text-align:left">AtSnRK2.6</td>
<td style="text-align:left">At4g33950</td>
</tr>
<tr>
<td style="text-align:left">AtSnRK2.7</td>
<td style="text-align:left">At4g40010</td>
</tr>
<tr>
<td style="text-align:left">AtSnRK2.8</td>
<td style="text-align:left">At1g78290</td>
</tr>
<tr>
<td style="text-align:left">AtSnRK2.9</td>
<td style="text-align:left">At2g23030</td>
</tr>
<tr>
<td style="text-align:left">AtSnRK2.10</td>
<td style="text-align:left">At1g60940</td>
</tr>
</tbody>
</table>
<table>
<thead><tr>
<th style="text-align:left">Gene Name</th>
<th style="text-align:left">Gene ID</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left">HvSnRK1.1</td>
<td style="text-align:left">HORVU1Hr1G081310</td>
</tr>
<tr>
<td style="text-align:left">HvSnRK1.2</td>
<td style="text-align:left">HORVU3Hr1G069190</td>
</tr>
<tr>
<td style="text-align:left">HvSnRK1.3</td>
<td style="text-align:left">HORVU3Hr1G107990</td>
</tr>
<tr>
<td style="text-align:left">HvSnRK1.4</td>
<td style="text-align:left">HORVU4Hr1G056610</td>
</tr>
<tr>
<td style="text-align:left">HvSnRK2.1</td>
<td style="text-align:left">HORVU1Hr1G055340</td>
</tr>
<tr>
<td style="text-align:left">HvSnRK2.2</td>
<td style="text-align:left">HORVU1Hr1G074670</td>
</tr>
<tr>
<td style="text-align:left">HvSnRK2.3</td>
<td style="text-align:left">HORVU2Hr1G029900</td>
</tr>
<tr>
<td style="text-align:left">HvSnRK2.4</td>
<td style="text-align:left">HORVU2Hr1G075470</td>
</tr>
<tr>
<td style="text-align:left">HvSnRK2.5</td>
<td style="text-align:left">HORVU2Hr1G110230</td>
</tr>
<tr>
<td style="text-align:left">HvSnRK2.6</td>
<td style="text-align:left">HORVU2Hr1G125950</td>
</tr>
<tr>
<td style="text-align:left">HvSnRK2.7</td>
<td style="text-align:left">HORVU3Hr1G082690</td>
</tr>
<tr>
<td style="text-align:left">HvSnRK2.8</td>
<td style="text-align:left">HORVU4Hr1G013540</td>
</tr>
<tr>
<td style="text-align:left">HvSnRK2.9</td>
<td style="text-align:left">HORVU5Hr1G018340</td>
</tr>
<tr>
<td style="text-align:left">HvSnRK2.10</td>
<td style="text-align:left">HORVU5Hr1G097630</td>
</tr>
<tr>
<td style="text-align:left">HvSnRK2.11</td>
<td style="text-align:left">HORVU0Hr1G011570</td>
</tr>
</tbody>
</table>
<p>使用 Mega7 分析完之后使用 <a href="https://itol.embl.de/">iTOL 网站</a> 进行美化：</p>
<p><img alt="" src="/i/20240427092499.jpg"/></p>
<h2>共线性分析</h2>
<p>共线性分析的步骤比较多，这里不展开赘述：</p>
<p><img alt="" src="/i/20240427092542.jpg"/></p>
<p><img alt="" src="/i/20240427092569.jpg"/></p>
<h2>多序列比对</h2>
<p>多序列比对我一般用 <a href="https://www.genome.jp/tools-bin/clustalw">CLUSTALW</a> 和 <a href="https://espript.ibcp.fr/ESPript/cgi-bin/ESPript.cgi">ESPript</a> 来做，将蛋白序列上传到 CLUSTALW 进行多序列比对，下载 aln 的结果后用 ESPript 进行可视化：</p>
<p><img alt="" src="/i/20240427093322.jpg"/></p>
<h2>蛋白质理化性质分析</h2>
<p>用 TBtools 即可一键分析，导入蛋白质序列即可：</p>
<p><img alt="" src="/i/20240427092592.jpg"/></p>
<table>
<thead><tr>
<th style="text-align:left">Sequence ID</th>
<th style="text-align:left">Number of Amino Acid</th>
<th style="text-align:left">Molecular Weight</th>
<th style="text-align:left">Theoretical pI</th>
<th style="text-align:left">Instability Index</th>
<th style="text-align:left">Aliphatic Index</th>
<th style="text-align:left">Grand Average of Hydropathicity</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left">OsSAPK1</td>
<td style="text-align:left">342</td>
<td style="text-align:left">38698.3</td>
<td style="text-align:left">5.43</td>
<td style="text-align:left">27.49</td>
<td style="text-align:left">89.47</td>
<td style="text-align:left">-0.208</td>
</tr>
<tr>
<td style="text-align:left">OsSAPK2</td>
<td style="text-align:left">339</td>
<td style="text-align:left">38538.14</td>
<td style="text-align:left">5.31</td>
<td style="text-align:left">32.18</td>
<td style="text-align:left">87.94</td>
<td style="text-align:left">-0.248</td>
</tr>
<tr>
<td style="text-align:left">OsSAPK3</td>
<td style="text-align:left">334</td>
<td style="text-align:left">37873.18</td>
<td style="text-align:left">5.67</td>
<td style="text-align:left">37.29</td>
<td style="text-align:left">85.87</td>
<td style="text-align:left">-0.424</td>
</tr>
<tr>
<td style="text-align:left">OsSAPK4</td>
<td style="text-align:left">360</td>
<td style="text-align:left">41978.75</td>
<td style="text-align:left">6.06</td>
<td style="text-align:left">53.65</td>
<td style="text-align:left">77.97</td>
<td style="text-align:left">-0.616</td>
</tr>
<tr>
<td style="text-align:left">OsSAPK5</td>
<td style="text-align:left">370</td>
<td style="text-align:left">42165.77</td>
<td style="text-align:left">5.99</td>
<td style="text-align:left">42.18</td>
<td style="text-align:left">77.22</td>
<td style="text-align:left">-0.501</td>
</tr>
<tr>
<td style="text-align:left">OsSAPK6</td>
<td style="text-align:left">365</td>
<td style="text-align:left">41803.36</td>
<td style="text-align:left">5.72</td>
<td style="text-align:left">43.13</td>
<td style="text-align:left">74.27</td>
<td style="text-align:left">-0.621</td>
</tr>
<tr>
<td style="text-align:left">OsSAPK7</td>
<td style="text-align:left">359</td>
<td style="text-align:left">41324.97</td>
<td style="text-align:left">5.83</td>
<td style="text-align:left">45.89</td>
<td style="text-align:left">77.41</td>
<td style="text-align:left">-0.567</td>
</tr>
<tr>
<td style="text-align:left">OsSAPK8</td>
<td style="text-align:left">371</td>
<td style="text-align:left">41723.44</td>
<td style="text-align:left">4.85</td>
<td style="text-align:left">44.33</td>
<td style="text-align:left">86.71</td>
<td style="text-align:left">-0.303</td>
</tr>
<tr>
<td style="text-align:left">OsSAPK9</td>
<td style="text-align:left">361</td>
<td style="text-align:left">40628.32</td>
<td style="text-align:left">4.81</td>
<td style="text-align:left">35.85</td>
<td style="text-align:left">85.62</td>
<td style="text-align:left">-0.272</td>
</tr>
<tr>
<td style="text-align:left">OsSAPK10</td>
<td style="text-align:left">362</td>
<td style="text-align:left">40699.27</td>
<td style="text-align:left">4.8</td>
<td style="text-align:left">40.36</td>
<td style="text-align:left">87.54</td>
<td style="text-align:left">-0.281</td>
</tr>
</tbody>
</table>
<h2>蛋白质二、三级结构预测</h2>
<p>原文作者使用 <a href="https://npsa-prabi.ibcp.fr/cgi-bin/npsa_automat.pl?page=npsa_sopma.html">SOPMA 在线网站</a> 进行蛋白质二级结构预测，直接提交序列就可以得到结果：</p>
<pre><code class="lang-txt">SOPMA :
   Alpha helix     (Hh) :   122 is  35.99%
   310  helix       (Gg) :     0 is   0.00%
   Pi helix        (Ii) :     0 is   0.00%
   Beta bridge     (Bb) :     0 is   0.00%
   Extended strand (Ee) :    62 is  18.29%
   Beta turn       (Tt) :    22 is   6.49%
   Bend region     (Ss) :     0 is   0.00%
   Random coil     (Cc) :   133 is  39.23%
   Ambiguous states (?)  :     0 is   0.00%
   Other states         :     0 is   0.00%
</code></pre>
<p><img alt="" src="/i/20240427092543.jpg"/></p>
<p>我这里仅仅分析了 OsSAPK2 的二级结构。</p>
<p>三级结构用 <a href="https://www.uniprot.org/">UniProt 网站</a>，直接搜索蛋白的 ID 即可：</p>
<p><img alt="" src="/i/20240427092565.jpg"/></p>
<p>这里我也同样只分析了 OsSAPK2 的三级结构。</p>
<h2>互作蛋白网络</h2>
<p>互作蛋白网络使用 STRING 数据库完成，直接输入基因 ID 和物种即可：</p>
<p><img alt="" src="/i/20240427092509.jpg"/></p>
<p>将数据导出为 tsv 文件后，可以使用 Cytoscape 进行美化。现在已经更新，默认的图已经很漂亮了，此处就不进行美化了。</p>
<h2>顺式作用元件分析</h2>
<p>将上游 2000 的 cds 序列上传到 <a href="http://bioinformatics.psb.ugent.be/webtools/plantcare/html/">plantcare 网站</a>，拿到结果后处理一下数据然后导入 TBtools 就可以进行绘图。</p>
<p><img alt="" src="/i/20240427092675.jpg"/></p>
<p><img alt="" src="/i/20240427092627.jpg"/></p>
<h2>表达热图</h2>
<p>原文的表达谱数据是从 <a href="https://ricexpro.dna.affrc.go.jp/data-set.html">CREP 数据库</a> 下载的，同样从 CREP 数据库下载表达量数据后用 TBtools 绘制热图即可：</p>
<p><img alt="" src="/i/20240427092647.jpg"/></p>
<h2>数据可用性</h2>
<p>本文使用的所有数据均已保存至 GitHub：<a href="https://github.com/imjiaoyuan/bio-code/tree/main/OsSnRK2-GFI">https://github.com/imjiaoyuan/bio-code/tree/main/OsSnRK2-GFI</a></p>
