<h2><a href="https://github.com/yuanj82/yuanj.top/blob/master/content/posts/ä¸€æ¬¡å¤±è´¥çš„è®ºæ–‡å¤ç°.md">ä»“åº“æºæ–‡</a>ï¼Œ<a href="https://yuanj.top/posts/v7x0u7t">ç«™ç‚¹åŸæ–‡</a></h2>
<hr/>
<p>title: ä¸€æ¬¡å¤±è´¥çš„è®ºæ–‡å¤ç°
tags:</p>
<ul>
<li>"Genomics"</li>
<li>"Publications"
slug: v7x0u7t
date: 2024-03-09T17:21:37+08:00</li>
</ul>
<hr/>
<p>æœ€è¿‘åœ¨å­¦ä¹ å¤šç»„å­¦åˆ†æï¼Œäºæ˜¯å†³å®šä»è®ºæ–‡å¤ç°å…¥æ‰‹ï¼Œæ‰¾äº†ä¸€ç¯‡æ„Ÿè§‰å¯ä»¥çš„æ–‡ç«  <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10057503/">ã€ŠIntegrated ATAC-Seq and RNA-Seq Data Analysis to Reveal OsbZIP14 Function in Rice in Response to Heat Stressã€‹</a> å°±ç›´æ¥å¼€å¹²äº†ã€‚</p>
&lt;!--more--&gt;

<p>è¿™ç¯‡æ–‡ç« ç»“åˆäº† RNA-seq å’Œ ATAC-seq ä¸¤ç§ç»„å­¦åˆ†ææ–¹æ³•ï¼Œé‰´å®šå‡ºäº†ä¸‰ä¸ªæ°´ç¨»ä¸­çš„çƒ­èƒè¿«å“åº”åŸºå›  OsHSF7ã€OsbZIP14 å’Œ OsMYB2ï¼Œé€šè¿‡ä¸€ç³»åˆ—å®éªŒè¯å®äº† OsbZIP14 åœ¨çƒ­èƒè¿«ä¸‹æ¤ç‰©ç”Ÿé•¿å‘è‚²ä¸­å‘æŒ¥ä½œç”¨ï¼Œæ­ç¤ºäº† OsbZIP14 åŸºå› ä¸çƒ­åº”æ¿€è¯±å¯¼çš„å¶ç‰‡æ—©è¡°ä¹‹é—´çš„å…³è”ï¼Œæ­¤å¤–ï¼Œä½œè€…è¿˜æ‰¾å‡ºå¹¶è¯å®äº† OsbZIP14 çš„ä¸€ä¸ªç›¸äº’ä½œç”¨è›‹ç™½ OsbZIP58ã€‚</p>
<p>æˆ‘ä¸»è¦å¯¹å…¶ä¸­çš„ RNA-seq å’Œ ATAC-seq ä¸¤ç§åˆ†æè¿›è¡Œå¤ç°ã€‚</p>
<h2>RNA-seq</h2>
<p>ä½œè€…ä½¿ç”¨çš„æµ‹åºæ•°æ®æ¥è‡ª NCBI SRA æ•°æ®åº“ï¼ŒBioProject ç¼–å·ä¸º PRJNA604026ï¼Œæ­¤é¡¹ç›®å…±æœ‰ 12 ä¸ªæ ·æœ¬çš„æ•°æ®ï¼Œä½†æ˜¯ä½œè€…åªç”¨äº† 6 ä¸ªï¼Œæˆ‘ä½¿ç”¨æ‰€æœ‰çš„ 12 ä¸ªæ ·æœ¬è¿›è¡Œåˆ†æã€‚RNA-seq åˆ†ææˆ‘å·²ç»æ˜¯æ¯”è¾ƒç†Ÿæ‚‰äº†ï¼Œä¹‹å‰ä¹Ÿå†™è¿‡ <a href="https://yuanj.top/tags/genomics/">ç›¸å…³çš„æ•™ç¨‹</a>ï¼Œäºæ˜¯ç™»å½•è¶…ç®—ï¼ŒæŒ‰ç…§ä½œè€…ææ–™ä¸æ–¹æ³•ä¸­çš„æ­¥éª¤ï¼ˆå¤§ä½“ç›¸åŒï¼‰ï¼Œfastqc è¿›è¡Œè´¨é‡æ£€æµ‹åä½¿ç”¨ Trim Galore å»é™¤æ¥å¤´è¿æ¥å­å’Œä½è´¨é‡åºåˆ—ï¼Œä½¿ç”¨ RAP DB ä¸Šçš„å‚è€ƒåŸºå› ç»„æ„å»º hisat2 ç´¢å¼•åï¼Œå°† reads æ¯”å¯¹åˆ°å‚è€ƒåŸºå› ç»„ï¼Œç„¶å samtools æ’åºã€å‹ç¼©ã€è½¬ä¸º bam æ ¼å¼ï¼Œæœ€åè¿™é‡Œä½œè€…æ²¡æœ‰å†™ç”¨ä»€ä¹ˆè½¯ä»¶è¿›è¡Œè®¡é‡ï¼Œåªå†™äº†<code>Clean reads were then mapped to the reference genome using HISAT2 v2.2.1, sorted using samtools v1.9, and FPKM was calculated for each gene.</code> æ‰€ä»¥æˆ‘ä¹Ÿå°±æŒ‰ç…§ä»¥å‰çš„å¥—è·¯ç”¨ subreadï¼Œæ‹¿åˆ°åŸºå› è®¡æ•°è¡¨æ ¼ä¹‹åç”¨ R è¯­è¨€è¿›è¡Œ FPKM æ ‡å‡†åŒ–ï¼š</p>
<pre><code class="lang-r">rm(list = ls())

counts &lt;- read.csv(
    'rna-seq.counts',
    header = TRUE,
    sep = '\t',
    # row.names = "Geneid",
    comment.char = '#',
    check.names = FALSE
)
# å¯¹ç¬¬ 6 åˆ—ä»¥åçš„æ•°æ®è¿›è¡Œ fpkm è®¡ç®—
for (clm in colnames(counts)[6:ncol(counts)]) {
    col_fpkm &lt;- paste0(clm, "_FPKM")     # æ–°åˆ—çš„åç§°ï¼ŒåŠ ä¸Š"_FPKM"åç¼€
    total &lt;- sum(counts[, clm])          # è®¡ç®—æ¯ä¸ªæ ·æœ¬çš„æ€»è¯»æ•°
    counts[col_fpkm] &lt;- (counts[, clm] * 10^6) / (counts[, "Length"] / 1000)  # ä½¿ç”¨ç›¸åº”æ ·æœ¬çš„é•¿åº¦å€¼è®¡ç®— FPKM å¹¶æ·»åŠ  FPKM åˆ— # nolint
}
# åˆ æ‰åŸæœ‰çš„ counts
counts = counts[,-c(2:19)]

numeric_mask &lt;- sapply(counts, is.numeric)
counts[numeric_mask] &lt;- lapply(counts[numeric_mask], function(x) ifelse(x &lt; 1, x + 1, x)) # nolint

write.table(counts, file = 'fpkm_output', sep = '\t', row.names = FALSE)
</code></pre>
<p>éšåå°±æ˜¯å·®å¼‚è¡¨è¾¾åˆ†æï¼Œè¿˜æ˜¯ Deseq2 åŒ…ï¼š</p>
<pre><code class="lang-r">rm(list = ls())  
Sys.setenv(LANGUAGE = "en")

library(DESeq2)

fpkm = read.csv(
    'fpkm_output', 
    header = T,  
    sep = '\t', 
    row.names = "Geneid", 
    comment.char = '#', 
    check.name = F
)

fpkm &lt;- round(fpkm)

numeric_mask &lt;- sapply(fpkm, is.numeric)
fpkm[numeric_mask] &lt;- lapply(fpkm[numeric_mask], function(x) ifelse(is.numeric(x) &amp; x &lt; 1, x + 100, x))

fpkm[-1, ] &lt;- apply(fpkm[-1, ], 2, as.integer)

missing_values &lt;- sum(is.na(fpkm))
if (missing_values &gt; 0) {
  fpkm &lt;- na.omit(fpkm)
}

samples = data.frame(
    sampleID = c("Normal-1-1_FPKM", "Normal-1-2_FPKM", "Normal-1-3_FPKM", "HS-1-1_FPKM", "HS-1-2_FPKM", "HS-1-3_FPKM", "Normal-2-1_FPKM", "Normal-2-2_FPKM", "Normal-2-3_FPKM", "HS-2-1_FPKM", "HS-2-2_FPKM", "HS-2-3_FPKM"), 
    sample = c("sample1", "sample1", "sample1", "sample2", "sample2", "sample2", "sample3", "sample3", "sample3", "sample4", "sample4", "sample4")
)
rownames(samples) = samples$sampleID
samples$sample = factor(samples$sample, levels = c('sample1', 'sample2', 'sample3', 'sample4'))
dds = DESeqDataSetFromMatrix(countData = fpkm, colData = samples, design = ~sample)
dds_count &lt;- DESeq(dds, fitType = 'mean', minReplicatesForReplace = 7, parallel = FALSE)

sampl1_vs_sample2 &lt;- results(dds_count, contrast = c('sample', 'sample1', 'sample2'))
sampl3_vs_sample4 &lt;- results(dds_count, contrast = c('sample', 'sample3', 'sample4'))

result1 &lt;- data.frame(sampl1_vs_sample2, stringsAsFactors = FALSE, check.names = FALSE)
result2 &lt;- data.frame(sampl3_vs_sample4, stringsAsFactors = FALSE, check.names = FALSE)

write.table(result1, 'sampl1_vs_sample2.DESeq2.txt', col.names = NA, sep = '\t', quote = FALSE)
write.table(result2, 'sampl3_vs_sample3.DESeq4.txt', col.names = NA, sep = '\t', quote = FALSE)
</code></pre>
<p>æŠŠç»“æœå¯è§†åŒ–åæˆ‘ä¸€çœ‹ï¼Œè·Ÿä½œè€…çš„ç›¸å·®çš„å¤ªå¤§äº† ... è€Œè¿™é‡Œè®¾ç½®çš„ p å€¼å’Œ log2FC çš„é˜ˆå€¼éƒ½æ˜¯ä¸ä½œè€…ä¸€è‡´çš„ã€‚</p>
<p><img alt="" src="https://images.yuanj.top/202403091748006.png"/></p>
<p>ä¸ä»…å¦‚æ­¤ï¼Œé‰´å®šå‡ºçš„å·®å¼‚åŸºå› æ•°é‡ä¹Ÿæœ‰å¾ˆå¤§çš„å·®åˆ«ï¼Œä½œè€…æ˜ç¡®è¯´äº†<code>In total, 24283 genes were identified in both the heat stress treated and control samples. To identify the critical functional genes related to heat stress resistance, differentially expressed genes (DEGs) were filtered based on the criteria |log2(fold change)| â‰¥ 1 and p-Value &lt; 0.05. Overall, 1942 upregulated and 3617 downregulated genes were screened.</code></p>
<p>ä½†æ˜¯æˆ‘é‰´å®šå‡ºçš„å·®å¼‚åŸºå› æœ‰ 34026 ä¸ªï¼Œä¸Šè°ƒä¸ä¸‹è°ƒçš„åŸºå› æ•°é‡ä¹Ÿå·®åˆ«å¾ˆå¤§ã€‚</p>
<p>æˆ‘çŒœæµ‹å¤±è´¥çš„åŸå› æœ‰ä»¥ä¸‹å‡ ä¸ªï¼š</p>
<ul>
<li>è®¡é‡æ—¶æ‰€ç”¨çš„è½¯ä»¶ä¸åŒ</li>
<li>æµ‹åºæ•°æ®é¢„å¤„ç†æ—¶å‚æ•°è®¾ç½®ä¸åŒ</li>
<li>å‚è€ƒåŸºå› ç»„å˜åŒ–ï¼Œä½œè€…æ˜¯ 2021 å¹´ 9 æœˆ 10 æ—¥è®¿é—®çš„ï¼Œå·²ç»è¿‡å»äº†ä¸¤å¹´å¤š</li>
<li>åŸºå› æ³¨é‡Šå˜åŒ–ï¼ŒRAP DB ä¸Šæœ€æ–°çš„æ˜¯ 2024 å¹´ 1 æœˆ 11 æ—¥æ›´æ–°çš„</li>
</ul>
<h2>ATAC-seq</h2>
<p>ATAC-seq çš„æ•°æ®ä¹Ÿæ˜¯æ¥è‡ª NCBI SRA æ•°æ®åº“ï¼ŒBioProject ç¼–å·ä¸º PRJNA604028ï¼Œæˆ‘æ˜¯ç¬¬ä¸€æ¬¡åšï¼Œä¹Ÿæ˜¯æŒ‰ç…§æ–‡ç« ææ–™ä¸æ–¹æ³•éƒ¨åˆ†å†™çš„<code>Using a similar approach to the RNA-seq analysis, data quality control, filtering, and comparison were performed.</code>æ‹¿åˆ° bam æ–‡ä»¶åä½¿ç”¨ macs2 è¿›è¡Œ Peak Callingï¼Œè¿™é‡Œ macs2 éœ€è¦æŒ‡å®šåŸºå› ç»„çš„å¤§å°ï¼Œæˆ‘æŒ‡å®šä¸º 3e8ï¼ˆæ°´ç¨»åŸºå› ç»„ä¸º 389Mï¼‰ï¼ŒæŠ¥é”™æ— æ³•è¾“å‡ºç»“æœ ... è¯•äº†å¾ˆå¤šä¸ªæ•°å­—éƒ½ä¸è¡Œï¼Œä¹Ÿæ²¡æœ‰æŸ¥åˆ°ç›¸å…³çš„èµ„æ–™ï¼Œäºæ˜¯æˆ‘å–æ¶ˆæŒ‡å®šåŸºå› ç»„å¤§å°ï¼Œå€’æ˜¯å¯ä»¥è¾“å‡ºç»“æœäº†</p>
<p>ç„¶åä½¿ç”¨ ChIPseeker è¿›è¡Œæ³¨é‡Šï¼š</p>
<pre><code class="lang-r">rm(list = ls())  

library(GenomicFeatures)
library(ChIPseeker)

spompe &lt;- makeTxDbFromGFF('transcripts.gff')
setwd('/home/yuanj/work/ata-seq/macs2_bed')
peak &lt;- readPeakFile('IRGSP_genome_57.bed')

peakAnno &lt;- annotatePeak(peak, tssRegion = c(-500, 500), TxDb = spompe)
</code></pre>
<p>annotatePeak å‡½æ•°ä¸€ç›´æŠ¥é”™<code>Error: invalid subscript</code>ï¼Œæ— è®ºæ€ä¹ˆæ”¹éƒ½ä¸è¡Œ ... äºæ˜¯æˆ‘çŒœæµ‹åº”è¯¥æ˜¯ macs2 è¾“å‡ºç»“æœçš„é—®é¢˜ã€‚</p>
<p>å†å°è¯•ä½¿ç”¨ homer è¿›è¡Œ motif æ³¨é‡Šä¸åˆ†æï¼Œä¹Ÿæ˜¯ä¸€ç›´æŠ¥é”™ ...</p>
<pre><code class="lang-bash">findMotifsGenome.pl IR57.bed tair10 motif -len 8,10,12
</code></pre>
<p>çœ‹æ¥åº”è¯¥æ˜¯ macs2 è¿™é‡Œå‡ºé”™äº†ï¼Œæš‚æ—¶è¿˜æ²¡æœ‰æ‰¾åˆ°è§£å†³åŠæ³•ï¼Œäºæ˜¯æ­¤æ¬¡è®ºæ–‡å¤ç°å°±è¢«è¿«ä¸­æ­¢äº†ğŸ˜«</p>
